elephant/test/test_phase_analysis.py

```diff
@@ -370,5 +373,266 @@ def testPhaseLockingValue_raise_Error_if_trial_lengths_are_different(self):
+            Time length of the datasets in seconds.
+        srate: float
+            Sampling rate used to 'record' the signals in Hz.
+
+        Returns
+        -------
+        None
+
+        Notes
+        -----
+        Used versions of MATLAB & FieldTrip:
+        MATLAB          Version 9.9         (2020b)
+        FieldTrip       fieldtrip-20210128
+
+        Instead of using the ft_connectivityanalysis() wrapper function, which
+        expects preprocessed data, the ft_connectivity_wpli() is called
+        directly. That's because the preprocessing functions of FieldTrip have
+        no 'conventional' Fourier-Transformation (FFT) method, but among
+        others a multitaper-FFT. Because this python-version of WPLI doesn't
+        use multitaper, but conventional FFT, the utilized MATLAB script also
+        uses the conventional FFT to preprocess the data and calculate the
+        cross-spectrum, which will then be passed to the ft_connectivity_wpli.
+
+        """
+        times = np.arange(0, tlength, 1 / srate)
+
+        # lfps_1 & 2 will be used for
+        # 1) calculating ground-truth with FieldTrips' ft_connectivity_wpli
+        # 2) comparison to mutlitaper-approaches like
+        # FieldTrips' ft_connectivityanalysis()
+        # and MNEs' spectral_connectivity at certain frequencies
+        lfps_1 = [np.cos(2 * 16 * np.pi * times + np.pi/2) +
+                  np.cos(2 * 36 * np.pi * times + np.pi / 2) +
+                  np.cos(2 * 52 * np.pi * times) +
+                  np.cos(2 * 100 * np.pi * times) +
+                  np.cos(2 * 70 * np.pi * times + np.pi/2 + np.pi*(i % 2)) +
+                  np.random.normal(loc=0.0, scale=1, size=len(times))
+                  for i in range(ntrial)]
+        lfps_1 = np.stack(lfps_1, axis=0)
+
+        lfps_2 = [np.cos(2 * 16 * np.pi * times) +
+                  np.cos(2 * 36 * np.pi * times) +
+                  np.cos(2 * 52 * np.pi * times + np.pi/2) +
+                  np.cos(2 * 100 * np.pi * times + np.pi/2) +
+                  np.cos(2 * 70 * np.pi * times) +
+                  np.random.normal(loc=0.0, scale=1, size=len(times))
+                  for _ in range(ntrial)]
+        lfps_2 = np.stack(lfps_2, axis=0)
+
+        # save artificial LFP-dataset to .mat files
+        mdic_1 = {"lfp_matrix": lfps_1, "time": times, "sf": srate}
+        mdic_2 = {"lfp_matrix": lfps_2, "time": times, "sf": srate}
+        filename1_a = os.path.sep.join(['cross_testing_scripts',
+                                        'artificial_LFPs_1.mat'])
+        filename2_a = os.path.sep.join(['cross_testing_scripts',
+                                        'artificial_LFPs_2.mat'])
+
+        savemat(filename1_a, mdic_1)
+        savemat(filename2_a, mdic_2)
+
+    def setUp(self):
+        self.tolerance = 1e-15
+
+        # simple samples of different shapes to assert ErrorRaising
+        self.simple_x = np.array([[0, -np.pi, np.pi], [0, -np.pi, np.pi]])
```

Comment:
To understand the tests later, more descriptive names can be used, like `trials2_length3`, `trials1_length3`, `trials1_length4`). This makes it easier to understand the two unit tests later, and also makes sure that `x`, `y`, or `z` are not swapped and you are indeed emulating that particular error in the test. 

Or, in order to improve simplicity and readability, you could instead implement just a single test to check the error of wrong input dimensions. Therefore, these arrays would be defined inside that test function and the two assertions for either violation would be tested at once. These variables are defined at the class set up just for those two small tests at the end.